### ISSUE TITLE
 CANCELLED

### ISSUE THREAD

**Issue by morsagmon** [capacitor-community/contacts#112]

**Comment by morsagmon** [comment]
Issue cancelled

**Comment by tafelnl** [comment]
I'm delighted that you seem to have managed to resolve the issue. Can you confirm if it was either a bug in this plugin or something else

**Comment by morsagmon** [comment]
Hi @tafelnl .
I cancelled it as I realized I misinterpreted the issue. 
After deeper investigation I found out that getContacts() is not consistent with the objects it returns.
For some contacts, the array member returned looks like this:
{conatcId: "4380"}
For others, it returns something like this:
{conatcId: "4330", names: {}}
And for others it returns:
{conatcId: "4330", names: {display: "first last", ...}, phones: {...}}
And some reflect the projection, while others return many other attributes not included in the projection, so the projection is not always respected.
So, these differences need to be handles by us after fetching the result set.
I do not have the tools to figure out what causes these differences in the contact record itself.

The original issue as I described here was about the contacts being displayed on one Android device and not on another, which turned out to be not the root of the issue.
